aicsimageio/readers/glob_reader.py

```diff
@@ -0,0 +1,384 @@
+    channel_names: Optional[Union[List[str], List[List[str]]]]
+        A list of string channel names to be applied to all array(s) or a
+        list of lists of string channel names to be mapped onto the list of arrays
+        provided to image.
+        Default: None (create OME channel IDs for names for single or multiple arrays)
+    single_file_shape : Optional[Tuple]
+        Expected shape for a single file of the set. If not provided, will attempt to
+        determine the shape from the first file found in the glob.
+        Default : None
+    single_file_dims : Optional[Tuple]
+        Dimensions that correspond to the data dimensions of a single file in the glob.
+        Default : ('Y', 'X')
+    """
+
+    @staticmethod
+    def _is_supported_image(fs: AbstractFileSystem, path: str, **kwargs: Any) -> bool:
+        try:
+            with fs.open(path) as open_resource:
+                with TiffFile(open_resource):
+                    return True
+
+        except (TiffFileError, TypeError):
+            return False
+
+    def __init__(
+        self,
+        glob_in: Union[str, List[str]],
+        indexer: Union[pd.DataFrame, Callable] = None,
+        chunk_dims: Union[str, List[str]] = DEFAULT_CHUNK_DIMS,
+        dim_order: Optional[Union[List[str], str]] = None,
+        channel_names: Optional[Union[List[str], List[List[str]]]] = None,
+        single_file_shape: Optional[Tuple] = None,
+        single_file_dims: Optional[Tuple] = (
+            DimensionNames.SpatialY,
+            DimensionNames.SpatialX,
+        ),
+        **kwargs: Any,
+    ):
+
+        # Assmble glob list if given a string
+        if isinstance(glob_in, str):
+            file_series = pd.Series(glob.glob(glob_in))
+        elif isinstance(glob_in, list):
+            file_series = pd.Series(glob_in)  # pd.DataFrame({"filename": glob_in})
+
+        if len(file_series) == 0:
+            raise ValueError("No files found matching glob pattern")
+
+        if indexer is None:
+            series_idx = [
+                DimensionNames.Samples,
+                DimensionNames.Time,
+                DimensionNames.Channel,
+                DimensionNames.SpatialZ,
+            ]
+            indexer = lambda x: pd.Series(
+                re.findall(r"\d+", Path(x).name), index=series_idx
+            ).astype(int)
+
+        if callable(indexer):
+            self._all_files = file_series.apply(indexer)
+            self._all_files["filename"] = file_series
+        elif isinstance(indexer, pd.DataFrame):
+            self._all_files = indexer
+            self._all_files["filename"] = file_series
+
+        sort_order = []
+        for dim in DEFAULT_DIMENSION_ORDER_LIST_WITH_SAMPLES:
+            if dim not in self._all_files.columns and dim not in single_file_dims:
+                self._all_files[dim] = 0
+            if dim in self._all_files.columns:
+                sort_order.append(dim)
```

Comment:
Can you explain this sorting in a comment? To my understanding its:

"Set the column in the all files dim to zero if the dim doesn't exist on the file, otherwise, if the dim is present, then add it to the sort order. Because we are using the default dimension ordering, this will naturally create a sort order based off the standard dimension order."